exoscale_database: resource implementation overhaul

falzm opened the pull request:
This change overhauls the `exoscale_database` resource implementation
following Exoscale API changes.

greut commented:
@JessicaToscani there is a new example to run and play with this change.

greut commented:
@LaCap if you could test it, that would be awesome.

greut commented:
@LaCap how can I help you help me?

greut commented:
@LaCap this one should work better, thanks!

https://ppsos-ch-dk-2.exo.io/yoanyoanyoan/terraform-provider-exoscale_v0.9.25_xfd0ce94.bz2

LaCap commented:
@greut apparently not:

```
blah examples/elastic-ip ‹associate-ip-id*› » terraform plan  
var.key_pair
  Enter a value: ...


Warning: provider.exoscale: "profile": [DEPRECATED] Use region instead



Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

exoscale_ipaddress.ingress: Refreshing state... (ID: c2d51905-9157-49ca-abf4-57916199a321)
exoscale_compute.machine: Refreshing state... (ID: f5a43466-524f-462c-a839-18d1ba57fc86)
exoscale_secondary_ipaddress.machine: Refreshing state... (ID: f91d7889-9cb8-48f4-8003-7f67b8a6389a)

Error: Error refreshing state: 1 error(s) occurred:

* exoscale_secondary_ipaddress.machine: 1 error(s) occurred:

* exoscale_secondary_ipaddress.machine: exoscale_secondary_ipaddress.machine: API error ParamError 431 (ServerApiException 9999): Unable to execute API command listnics due to invalid value. Invalid parameter nicid value=e4a57c66-3be8-48fe-a47a-518f9dd5ee20 due to incorrect long value format, or entity does not exist or due to incorrect parameter annotation for the field in api cmd class.
```
Or maybe I'm doing something wrong.

greut commented:
@LaCap thanks! You may rename the _profile_ into _region_ in `main.tf`.

https://ppsos-ch-dk-2.exo.io/yoanyoanyoan/terraform-provider-exoscale_v0.9.25_xac48aa9.bz2

greut commented:
@LaCap jamais deux sans trois.

https://ppsos-ch-dk-2.exo.io/yoanyoanyoan/terraform-provider-exoscale_v0.9.25_x179ad5b.bz2

LaCap commented:
@greut nope...
Still the same error:

```
* exoscale_secondary_ipaddress.machine: exoscale_secondary_ipaddress.machine: API error InternalError 530 (ServerApiException 9999): Virtual machine id does not exist
```

greut commented:
run the example, destroy the machine, refresh = :boom: ...

greut commented:
@LaCap yet another one :-)

https://ppsos-ch-dk-2.exo.io/yoanyoanyoan/terraform-provider-exoscale_v0.9.25_x2c50642.bz2

LaCap commented:
@greut :+1: works better now.

greut commented:
thanks and sorry for the too many try'n'fail

Commit:
exoscale_database: resource implementation overhaul

This change overhauls the `exoscale_database` resource implementation
following Exoscale API changes.

Commit:
Fix acceptance tests
